packages/core/database/src/metadata/relations.ts

```diff
@@ -417,6 +425,10 @@ const createJoinTable = (
   metadata: Metadata,
   { attributeName, attribute, meta }: JoinTableOptions
 ) => {
+  if (!shouldUseJoinTable(attribute)) {
```

Comment:
Just a note, every place in the codebase the calls createJoinTable has a check for this before it now. And I don't think we expose it. So this should help catch future mistakes.